src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs

```diff
@@ -586,31 +586,40 @@ public int LoadXSL_Resolver(String _strXslFile, XmlResolver xr)
         //VerifyResult
         public void VerifyResult(string expectedValue)
         {
-            XmlDiff.XmlDiff xmldiff = new XmlDiff.XmlDiff();
-            xmldiff.Option = XmlDiffOption.InfosetComparison | XmlDiffOption.IgnoreEmptyElement;
-
-            StreamReader sr = new StreamReader(new FileStream("out.xml", FileMode.Open, FileAccess.Read));
-            string actualValue = sr.ReadToEnd();
-            sr.Dispose();
-
-            //Output the expected and actual values
-            _output.WriteLine("Expected : " + expectedValue);
-            _output.WriteLine("Actual : " + actualValue);
-
-            //Load into XmlTextReaders
-            XmlTextReader tr1 = new XmlTextReader("out.xml");
-            XmlTextReader tr2 = new XmlTextReader(new StringReader(expectedValue));
-
-            bool bResult = xmldiff.Compare(tr1, tr2);
-
-            //Close the readers
-            tr1.Dispose();
-            tr2.Dispose();
+            lock(s_outFileMemoryLock)
+            {
+                XmlDiff.XmlDiff xmldiff = new XmlDiff.XmlDiff();
+                xmldiff.Option = XmlDiffOption.InfosetComparison | XmlDiffOption.IgnoreEmptyElement | XmlDiffOption.NormalizeNewline;
+                
+                string actualValue;
+                
+                using (FileStream outFile = new FileStream("out.xml", FileMode.Open, FileAccess.Read))
+                {
+                    StreamReader sr = new StreamReader(outFile);
```

Comment:
or replace all with File.ReadAllText

---
In reply to: [90154953](https://github.com/dotnet/corefx/pull/14113#discussion_r90154953) [](ancestors = 90154953)